### ISSUE TITLE
No option for max table width

### ISSUE THREAD

**Issue by idittansikte** [yzhang-gh/vscode-markdown#455]
#### What is the problem?
Not possible to set max width on table format.

Example, when formatting a table like this:

```text
| H1                | H2                                  | H3  |
| :------------- | :------------------------- | :---- |
| Bar                | Some very<br>long line of<br>text Some very<br>long line of<br>textSome very<br>long line of<br>text Some very<br>long line of<br>text | Foo |
```
it will format it into something like this:

```text
| H1                 | H2                                                                                                                                     | H3  |
| :----------------- | :------------------------------------------------------------------------------------------------------------------------------------- | :-- |
| Bar                | Some very<br>long line of<br>text Some very<br>long line of<br>textSome very<br>long line of<br>text Some very<br>long line of<br>text | Foo |
```

I would like a setting to set the max amount of dashes ("-" in "| :--- |") since Pandoc will use number of dashes to set the rendered table width in html.

**Comment by yzhang-gh** [comment]
Are you expecting something like this? (if up to 5 dashes)

```markdown
| H1   | H2                       | H3   |
| :--- | :-----                   | :--- |
| Bar  | Very long long long text | Foo  |
```

**Comment by idittansikte** [comment]
Precisely, would it be possible to add with reasonable effort? Another possible solution (and also another feature request) would be to add an option to disable formatting of the whole document and just format the table where the cursor is at.

Otherwise, this is an awesome plugin, thanks!

**Comment by idittansikte** [comment]
On second thought, pandoc might also count the spaces (in `| :----      |`), I will test this tomorrow. I have only tried not to pad with spaces.

**Comment by yzhang-gh** [comment]
Thanks for the reply.

It is doable. However, if we make the outcome like the later one, the formatting seems meaningless...

Does Pandoc have an option about the table width? That would be a better solution.

**Comment by idittansikte** [comment]
I've tried it out and spaces is not taken into account by Pandoc. However, I would prefer not to pad with spaces since the whole idea is to have a small/neat table in the markdown file and just a few rows that stand out from the rest of the table.

No, its not meaningless, I usually have a few tables that have one or two rows that are long and but I still want to format the rest of the tables.

Like this:
```
| H1   | H2     | H3   |
| :---- | :----- | :--- |
| Bar1  | Normal | Foo  |
| Bar2  | Very long long long text | Foo  |
| Bar3  | Normal | Foo  |
| Bar4  | Normal | Foo  |
```

If you don't think this is a good idea, can you at least consider an option to only format the table the cursor is at? Because now, all tables in the whole file are formatted when `Alt-Shift-F`. Then I could format the tables with long rows myself.

I don't think it's possible to set the width of tables in Pandoc in an easy way. Maybe with html tags and css inside the markdown but that is too much effort.

**Comment by yzhang-gh** [comment]
Thanks for the information.

I will leave this (option for max cell width) as a feature request. Let's see whether there are other people having the same issue.

> an option to only format the table the cursor is at

I will add the support for `DocumentRangeFormatting` later. (Probably next weekend as I am away this week)
